File: test_javaPrj_01/src/programmers/lv0/배열회전시키기.java
package src.programmers.lv0;

import java.util.ArrayList;
import java.util.List;

public class 배열회전시키기 {

    private 배열회전시키기(){}


    public void Solution() throws Exception{

        // int[] numbers = {1, 2, 3};
        int[] numbers = {4, 455, 6, 4, -1, 45, 6};

        String direction = "right";

        List<Integer> list = new ArrayList<>();

        for(int i = 0; i < numbers.length; i++){

            if("left".equals(direction)){
                if(i == numbers.length-1){
                    list.add(numbers[0]);
                }else{
                    list.add(numbers[i+1]);
                }
            }

            if("right".equals(direction)){
                if(i == 0){
                    list.add(numbers[numbers.length-1]);
                }else{
                    list.add(numbers[i-1]);
                }
            }
        }
        
        list.stream().mapToInt(o -> o).toArray();

        // List<Integer> list = new ArrayList<>();
        // for(int i = 0; i < numbers.length; i++){
        //     if(i == 0){

        //     }
        //     list.add(numbers[i]);
        //     if(i == numbers.length){

        //     }
        // }
        // System.out.println(Arrays.toString(answer));

    }

    public static void main(String[] args) {
        
        try{
            new 배열회전시키기().Solution();
        }catch(Exception e){
            e.printStackTrace();
        }


    }
    
}

File: test_javaPrj_01/src/programmers/lv0/배열회전시키기.py


class 배열회전시키기:

    def __init__(self, numbers:list[int], direction:str) -> None:
        print(numbers)
        print( [numbers[-1]] + numbers[:-1] )
        print( numbers[1:] + [numbers[0]] ) 
        
        # Python은 list간 연산 시 새로운 List 합으로 생성된다.
        # Python의 Slicing과 활용하여 원소 제거 후 합연산을 활용했다.
        # Extended Slices 추가적으로 참고 필요

        print( [numbers[-1]] + numbers[:-1] if direction == 'right' else numbers[1:] + [numbers[0]] )


배열회전시키기([4, 455, 6, 4, -1, 45, 6], 'left')
